contracts/MerkleDelegation.sol

```diff
@@ -98,4 +133,41 @@ contract MerkleDelegation is Ownable, Pausable {
     {
         _transferOwnership(newOwner);
     }
+
+    function verifyVotingPower(
+        address delegator,
+        address voter,
+        uint256 weight,
+        uint256 blockNumber,
+        bytes32[] memory proof
+    ) public view returns (bool) {
+        // between last merkle root and now we use the last merkle root.
+
+        // search for the appropriate merkle root (the next root with smaller block number).
+        // ie. if we have root for block numbers [10,55,123] then for block 60 we have to use the hash of block 55
+        // and for number 131 we use hash of block 123, and for block 9 (or 10) where is hash (with smaller #block)
+        // then it can be verified that delegated stake was zero, for this voter.
+
+        uint256 checkpoint = _getPrevCheckpoint(delegator, blockNumber);
+        return
+            proof.verify(
+                delegation[delegator][checkpoint].trieRoot,
+                keccak256(abi.encodePacked(voter, weight, governanceToken))
+            );
+    }
+
+    function _getPrevCheckpoint(address delegator, uint256 _blockNum)
+        internal
+        view
+        returns (uint256 blockNum)
+    {
+        //TODO: do Binary Search instead of Linear Search.
```

Comment:
issue: TODOs in main are a bad habit and suggest an incomplete PR (as there is still a TODO).

Remove the TODO and create a Linear Issue on the backlog for prioritsation, if the TODO does not need to be done for this unit of work.

Reply:
gotcha

Reply:
moved 2 TODOs to Linear.